### ISSUE TITLE
Incorrect syntax near '='.

### ISSUE THREAD

**Issue by MajidSafari** [linq2db/linq2db#2652]
Hi when query

      where (deptCode != 4 && deptCode != 5 ? g.Sum(r => r.ExceptCount) > 50 : g.Sum(r => r.ExceptCount) > 1)

TSQL

HAVING
	(Sum([r].[ExceptCount]) > 50) = 1

what this is "= 1" ???????

Incorrect syntax near '='.

### Environment details
linq2db version: 3.1.6
Database Server: 2017
Database Provider: sql
Operating system: Win
.NET Framework: 4.8

**Comment by MajidSafari** [comment]
tanx @sdanyliv

**Comment by sdanyliv** [comment]
Why closed?

**Comment by MajidSafari** [comment]
Oh sorry ,

**Comment by MaceWindu** [comment]
Could you add whole query? I cannot reproduce it with 3.1.6:
```sql
HAVING
	IIF(@p1 = 1, IIF(Sum([t1].[ParentID]) > 50, 1, 0), IIF(Sum([t1].[ParentID]) > 1, 1, 0)) = 1
```

**Comment by MaceWindu** [comment]
BTW, it is possibly fixed with 3.2.0 as we introduced massive predicates refactoring there and in 3.2.0 I get
```sql
HAVING
	Sum([t1].[ParentID]) > 50
```
instead if one above

**Comment by MajidSafari** [comment]
Yes, I update to Release 3.2.1

I think fixed
